tests/cannon/evaluation-order1.dora

```diff
@@ -0,0 +1,15 @@
+//= stdout "Foo\nsomeInt\nbar\n"
```

Comment:
Can this test already be executed with Cannon? Maybe we could use `//= cannon` here

Reply:
I did't add `//= cannon`, but added `@cannon` to the methods. Otherwise it would break the "convention" of the tests in `tests/cannon`, which are all without the test annotation.
If it makes sense to run it twice with the baseline compiler and cannon, I could move it outside of the folder?

Reply:
Yes, I think it would make sense to run the test with both compilers. Just to ensure that their behavior matches.